provider/core/src/resource.rs

```diff
@@ -110,30 +110,30 @@ unsafe impl EqULE for ResourceKeyHash {}
 /// # use icu_provider::prelude::ResourceKey;
 /// const K: ResourceKey = icu_provider::resource_key!("foo/../bar@1");
 /// ```
-#[derive(PartialEq, Eq, Copy, Clone, PartialOrd, Ord)]
+#[derive(Copy, Clone)]
 pub struct ResourceKey {
     // This string literal is wrapped in leading_tag!() and trailing_tag!() to make it detectable
     // in a compiled binary.
     path: &'static str,
     hash: ResourceKeyHash,
 }
 
-// I don't think this lint makes sense, see https://github.com/rust-lang/rust-clippy/issues/2627.
-// It's not possible to violate the hash invariant (k1 == k2 -> hash(k1) == hash(k2)) without
-// using non-determinism or unsafe code when PartialEq is derived.
-#[allow(clippy::derive_hash_xor_eq)]
+impl PartialEq for ResourceKey {
```

Comment:
These are related because changing the databake match to not match on `ResourceKey` lets us remove the derived `PartialEq` and instead do something more efficient.

Should we compare (hash, string)? Then we get a quick exit if the hashes don't match.

Reply:
> Should we compare (hash, string)? Then we get a quick exit if the hashes don't match.

That would address my concern